### PR TITLE
Added code for Open hashing in c

### PR DESCRIPTION
**Pull request by cintamg**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3548

### CODE REVIEW

**Comment by cintamg** [comment]
@ankitaggarwal23 
Why does it show merge conflict?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 
- Add a new line at the EOF
- Add function descriptions
- Resolve the merge conflicts
- Format the Code properly

**Comment by cintamg** [comment]
Hey, @HarshCasper @ankitaggarwal23 @anushkrishnav @siddharth25pandey , please do review my pr
Thank you.

File: C/ds/README.md [hunk 1]
```diff
@@ -1 +1 @@
-a
+- [Open Hashing](ds/Open_hashing.c)
```
Comment (HarshCasper) [inline C/ds/README.md]: Don't make the changes here. Make it on `C/README.md`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by cintamg** [comment]
@HarshCasper I have made the changes. Please do review my pr.

**Comment by cintamg** [comment]
Hey, @HarshCasper @ankitaggarwal23 @anushkrishnav @siddharth25pandey , please do review my pr
Thank you.

**Comment by cintamg** [comment]
@HarshCasper I request you to review PR
Thank You

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**